src/ray/common/cgroup/cgroup_utils.cc

```diff
@@ -0,0 +1,185 @@
+// Copyright 2024 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ray/common/cgroup/cgroup_utils.h"
+
+#include <sys/stat.h>
+
+#include <fstream>
+
+#include "absl/strings/str_format.h"
+#include "absl/strings/str_join.h"
+#include "absl/strings/str_split.h"
+#include "ray/util/logging.h"
+
+namespace ray {
+
+namespace {
+
+// Owner can read and write.
+constexpr int kCgroupV2FilePerm = 0600;
+
+// There're two types of memory cgroup constraints:
+// 1. For those with limit capped, they will be created a dedicated cgroup;
+// 2. For those without limit specified, they will be added to the default cgroup.
+static constexpr std::string_view kDefaultCgroupV2Uuid = "default_cgroup_uuid";
+
```

Comment:
It's not a uuid, and we should not use a name like this visible in linux. We had the idea of getting a default name from cluster ID right?

Reply:
Renamed as `id`.

Reply:
> We had the idea of getting a default name from cluster ID right?

I'm not sure how cluster id is related here?